app/controllers/architectures_controller.rb

```diff
@@ -39,4 +56,55 @@ def destroy
       process_error
     end
   end
+
+  private
+
+  def canEdit?(id)
```

Comment:
```suggestion
  def can_edit?(id)
```

This is ruby, after all ;-)